packages/core/node/core-node-server-internal/src/node_config.ts

```diff
@@ -6,33 +6,61 @@
  * Side Public License, v 1.
  */
 
-import { schema } from '@kbn/config-schema';
+import { schema, TypeOf } from '@kbn/config-schema';
 import type { ServiceConfigDescriptor } from '@kbn/core-base-server-internal';
 
 /** @internal */
 export const NODE_CONFIG_PATH = 'node' as const;
+/**
+ * Wildchar is a special config option that implies all {@link NODE_DEFAULT_ROLES} roles.
+ * @internal
+ */
+export const NODE_WILDCARD_CHAR = '*' as const;
+/** @internal */
+export const NODE_BACKGROUND_TASKS_ROLE = 'background_tasks' as const;
+/** @internal */
+export const NODE_UI_ROLE = 'ui' as const;
 /** @internal */
-export const NODE_WILDCARD_CHAR = '*';
+export const NODE_MIGRATOR_ROLE = 'migrator' as const;
 /** @internal */
-export const NODE_ACCEPTED_ROLES = ['background_tasks', 'ui'];
+export const NODE_DEFAULT_ROLES = [NODE_BACKGROUND_TASKS_ROLE, NODE_UI_ROLE] as const;
```

Comment:
This is making me regret adding wildcard support in the first place. It's a little weird that `*` doesn't actually represent _all_ roles, since the migrator role is exclusive and cannot be combined with others.

But despite that, I think your implementation here is the best path forward. Better to validate that `migrator` is only ever used on its own, since any other configuration would be incorrect.